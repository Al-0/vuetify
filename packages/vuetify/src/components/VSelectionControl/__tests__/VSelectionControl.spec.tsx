// Components
import { VSelectionControl } from '..'

// Utilities
import { createVuetify } from '@/framework'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from '@jest/globals'

describe('VSelectionControl', () => {
  const vuetify = createVuetify()

  function mountFunction (options = {}) {
    return mount({
      render: () => (<VSelectionControl ref="control" />),
    }, {
      global: { plugins: [vuetify] },
      ...options,
    })
  }

  it('should expose variables when using refs', () => {
    const wrapper = mountFunction()
    const vm = wrapper.findComponent({ ref: 'control' })
    const input = wrapper.find('input')

    expect(vm.componentVM.isFocused).toBe(false)

    input.trigger('focus')

    expect(vm.componentVM.isFocused).toBe(true)
  })
})
