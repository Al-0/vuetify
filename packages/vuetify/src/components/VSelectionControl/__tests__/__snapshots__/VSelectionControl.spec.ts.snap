// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSelectionControl should match a snapshot 1`] = `
<div class="v-selection-control v-selection-control--density-default">
  <div class="v-selection-control__wrapper">
    <div class="v-selection-control__input">
      <input id="input-0">
    </div>
  </div>
  <label class="v-label"
         for="input-0"
  >
  </label>
</div>
`;
